Lombiq.Tests.UI/Extensions/ReliabilityUITestContextExtensions.cs

```diff
@@ -209,4 +246,23 @@ public static class ReliabilityUITestContextExtensions
         context.Missing(elementToWaitForGoMissing.Safely().Within(
             existsTimeout,
             interval ?? context.Configuration.TimeoutConfiguration.RetryInterval));
+
+    private static string ComputeElementImageHash(this UITestContext context, IWebElement element)
+    {
+        using var elementImage = context.TakeElementScreenshot(element);
+        using var elementImageStream = new MemoryStream();
+
+        elementImage.Save(elementImageStream, ImageFormat.Bmp);
+        var elementImageRaw = elementImageStream.ToArray();
+        return ComputeSha256Hash(elementImageRaw);
+    }
+
+    private static string ComputeSha256Hash(byte[] raw)
+    {
+        using var sha256Hash = SHA256.Create();
+
+        return string.Concat(
+            sha256Hash.ComputeHash(raw)
+                .Select(item => item.ToString("x2", CultureInfo.InvariantCulture)));
```

Comment:
`elementImageStream.GetHashCode()` is not enough to check the equality. ([here ](https://docs.microsoft.com/en-us/dotnet/api/system.object.gethashcode?view=net-6.0#:~:text=Two%20objects%20that%20are%20equal%20return%20hash%20codes%20that%20are%20equal.%20However%2C%20the%20reverse%20is%20not%20true%3A%20equal%20hash%20codes%20do%20not%20imply%20object%20equality%2C%20because%20different%20(unequal)%20objects%20can%20have%20identical%20hash%20codes.) you can read more)

If I remember good (in FW3.5) the `GetHashCode(...)` didn't generated position sensitive hash.

Reply:
For the record, this is what I wanted to reply here from e-mail but that [was only added under the PR](https://github.com/Lombiq/UI-Testing-Toolbox/pull/166#issuecomment-1145829841):

In any case, please check how the number of allocations can be reduced here, because there's both a stream and an array here.